ubiconfig/_impl/loaders/gitlab.py

```diff
@@ -23,80 +23,101 @@ def __init__(self, url):
         self._url = url
         self._session = requests.Session()
         self._repo_api = RepoApi(self._url.rstrip("/"))
+        self._branches = self._get_branches()
         self._files_branch_map = self._pre_load()
 
-    def load(self, file_name):
+    def load(self, file_name, version=None):
         """ Load file from remote repository.
         :param file_name: filename that is on remote repository in any branch
         """
-        # find the right branch from the mapping
-        branch_sha1 = self._files_branch_map[file_name]
-        version = branch_sha1[0][3:]  # such as ubi7, ubi7.1
+        if file_name not in self._files_branch_map:
+            raise ValueError("Couldn't find file %s in remote repo" % file_name)
 
-        config_file_url = self._repo_api.get_file_content_api(file_name, branch_sha1[1])
-        LOG.info("Loading configuration file from remote: %s", file_name)
+        sha1 = self._branches.get(version)
+        if version and not sha1:
+            LOG.warning(
+                "Didn't find version %s, will try to find %s in default",
+                version,
+                file_name,
+            )
+
+        if not version or not sha1:
+            # branch is not available from the wanted version or not specified,
+            # use the default version.
+            for branch_sha1 in self._files_branch_map[file_name]:
+                if branch_sha1[0] == "ubi7":
```

Comment:
I don't like much hardcoding "ubi7" and "ubi8" strings here. What will happen with ubi9? :D
I always thought that fallback to default version should do caller of load() and not ubi-config itself.

Maybe also raising an exception can be correct in this situation (which makes version more less mandatory :) ).
 
What's your thoughts?
But for now it's not a big deal, I believe for loading from Gitlab ubipop so far used load_all() at least in jenkins jobs.

Reply:
> I don't like much hardcoding "ubi7" and "ubi8" strings here. What will happen with ubi9? 

I was thinking about using environment variables, instead of hardcoding it here, that should be enough. But I want to open an issue for this. 
> I always thought that fallback to default version should do caller of load() and not ubi-config itself.
I don't understand what do you mean here, do you mean, it's user's responsibility to load the default version again? if that's the case, then we don't even need to discuss the hardcode ubi7/8 problem. ?

Reply:
> Maybe also raising an exception can be correct in this situation (which makes version more less mandatory :) ).

Yeah, raise an exception could be a good option as well.

Reply:
> I don't understand what do you mean here, do you mean, it's user's responsibility to load the default version again? if that's the case, then we don't even need to discuss the hardcode ubi7/8 problem. ?

When I was designing that new branching scheme I guess this wasn't taken as a use case.
In practice load("some.yml") would match more than one file, if you search across all branches.
We should probably re-think behavior of GitlabLoader.load() or even LocalLoader.load()

In different issue/PR of course.